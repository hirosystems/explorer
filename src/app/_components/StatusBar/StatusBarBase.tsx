import React, { FC, ReactNode } from 'react';
import { BsExclamationCircle, BsExclamationTriangle } from 'react-icons/bs';
import { IncidentImpact } from 'statuspage.io';

import { PAGE_MAX_WIDTH } from '../../../common/constants/constants';
import { Box } from '../../../ui/Box';
import { Flex } from '../../../ui/Flex';
import { Icon } from '../../../ui/Icon';
import { getColor } from './utils';

export const StatusBarBase: FC<{ impact: IncidentImpact; content: ReactNode }> = ({
  content,
  impact,
}) => {
  const icon =
    impact === IncidentImpact.Critical ? (
      <Icon as={BsExclamationCircle} color={getColor(impact)} />
    ) : impact === IncidentImpact.None ? null : (
      <Icon as={BsExclamationTriangle} color={getColor(impact)} />
    );
  return (
    <Box
      position={'sticky'}
      width={'100%'}
      top={'0'}
      backdropFilter={'blur(10px)'}
      zIndex={'docked'}
    >
      <Box
        width={'100%'}
        background={'rgba(255, 255, 255, 0.8)'}
        display={'flex'}
        alignItems={'center'}
        justifyContent={'center'}
        padding={'20px 0'}
      >
        <Flex
          width={'100%'}
          maxWidth={PAGE_MAX_WIDTH}
          padding={'0 32px'}
          alignItems={'center'}
          justifyContent={'center'}
          gap={'20px'}
        >
          {icon}
          {content}
        </Flex>
      </Box>
    </Box>
  );
};
