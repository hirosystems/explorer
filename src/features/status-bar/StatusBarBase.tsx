import { ReactNode } from 'react';
import { IncidentImpact } from 'statuspage.io';
import { BsExclamationCircle, BsExclamationTriangle } from 'react-icons/bs';
import { Icon } from '@/ui/Icon';
import { getColor } from '@/features/status-bar/utils';
import { Box } from '@/ui/Box';
import { Flex } from '@/ui/Flex';
import { PAGE_MAX_WIDTH } from '@/common/constants';

export function StatusBarBase({ content, impact }: { impact: IncidentImpact; content: ReactNode }) {
  if (impact === IncidentImpact.None) return null;
  const icon =
    impact === IncidentImpact.Critical ? (
      <Icon as={BsExclamationCircle} color={getColor(impact)} />
    ) : (
      <Icon as={BsExclamationTriangle} color={getColor(impact)} />
    );
  return (
    <Box
      width="100%"
      background="rgba(255, 255, 255, 0.8)"
      display="flex"
      alignItems="center"
      justifyContent="center"
      padding="20px 0"
    >
      <Flex
        width="100%"
        maxWidth={PAGE_MAX_WIDTH}
        padding="0 32px"
        alignItems="center"
        justifyContent="center"
        gap="20px"
      >
        {icon}
        {content}
      </Flex>
    </Box>
  );
}
