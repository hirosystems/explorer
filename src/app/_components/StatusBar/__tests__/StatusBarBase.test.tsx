import React from 'react';
import renderer from 'react-test-renderer';
import { IncidentImpact } from 'statuspage.io';

import { StatusBarBase } from '../StatusBarBase';

describe('StatusBarBase', () => {
  it('renders correctly (critical impact)', () => {
    const tree = renderer
      .create(<StatusBarBase impact={IncidentImpact.Critical} content={<div>Test Content</div>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly (non-critical impact)', () => {
    const tree = renderer
      .create(<StatusBarBase impact={IncidentImpact.Minor} content={<div>Test Content</div>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
