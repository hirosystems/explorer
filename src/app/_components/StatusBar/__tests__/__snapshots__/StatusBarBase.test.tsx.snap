// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBarBase renders correctly (critical impact) 1`] = `
<div
  className="css-lv2dgf"
>
  <div
    className="css-8vwxnf"
  >
    <div
      className="css-g55u9n"
    >
      <svg
        className="chakra-icon css-1fimugm"
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
        />
        <path
          d="M7.002 11a1 1 0 1 1 2 0 1 1 0 0 1-2 0zM7.1 4.995a.905.905 0 1 1 1.8 0l-.35 3.507a.552.552 0 0 1-1.1 0L7.1 4.995z"
        />
      </svg>
      <div>
        Test Content
      </div>
    </div>
  </div>
</div>
`;

exports[`StatusBarBase renders correctly (non-critical impact) 1`] = `
<div
  className="css-lv2dgf"
>
  <div
    className="css-8vwxnf"
  >
    <div
      className="css-g55u9n"
    >
      <svg
        className="chakra-icon css-16jze27"
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.938 2.016A.13.13 0 0 1 8.002 2a.13.13 0 0 1 .063.016.146.146 0 0 1 .054.057l6.857 11.667c.036.06.035.124.002.183a.163.163 0 0 1-.054.06.116.116 0 0 1-.066.017H1.146a.115.115 0 0 1-.066-.017.163.163 0 0 1-.054-.06.176.176 0 0 1 .002-.183L7.884 2.073a.147.147 0 0 1 .054-.057zm1.044-.45a1.13 1.13 0 0 0-1.96 0L.165 13.233c-.457.778.091 1.767.98 1.767h13.713c.889 0 1.438-.99.98-1.767L8.982 1.566z"
        />
        <path
          d="M7.002 12a1 1 0 1 1 2 0 1 1 0 0 1-2 0zM7.1 5.995a.905.905 0 1 1 1.8 0l-.35 3.507a.552.552 0 0 1-1.1 0L7.1 5.995z"
        />
      </svg>
      <div>
        Test Content
      </div>
    </div>
  </div>
</div>
`;
